src/main/java/core/basesyntax/dao/CarJdbcDaoImpl.java

```diff
@@ -0,0 +1,170 @@
+package core.basesyntax.dao;
+
+import core.basesyntax.exception.DataProcessingException;
+import core.basesyntax.lib.Dao;
+import core.basesyntax.model.Car;
+import core.basesyntax.model.Driver;
+import core.basesyntax.model.Manufacturer;
+import core.basesyntax.util.ConnectionUtil;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class CarJdbcDaoImpl implements CarDao {
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String query = "SELECT * FROM cars c "
+                + "JOIN manufacturers m on m.manufacturer_id = c.manufacturer_id "
+                + "JOIN cars_drivers cd on c.car_id = cd.car_id "
+                + "JOIN drivers d on d.driver_id = cd.driver_id "
+                + "WHERE d.driver_id = ? AND c.deleted = FALSE AND d.deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement carsByDriverStatement = connection.prepareStatement(query)) {
+            carsByDriverStatement.setLong(1, driverId);
+            ResultSet result = carsByDriverStatement.executeQuery();
+            List<Car> cars = new ArrayList<>();
+            while (result.next()) {
+                cars.add(getCar(result, connection));
+            }
+            return cars;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t get car by driver id " + driverId, e);
+        }
+    }
+
+    @Override
+    public Car create(Car car) {
+        String query = "INSERT INTO cars(model, manufacturer_id) VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement createdCar = connection
+                        .prepareStatement(query, Statement.RETURN_GENERATED_KEYS)) {
+            createdCar.setString(1, car.getModel());
+            createdCar.setLong(2, car.getManufacturer().getId());
+            createdCar.executeUpdate();
+            ResultSet resultSet = createdCar.getGeneratedKeys();
+            if (resultSet.next()) {
+                car.setId(resultSet.getObject(1, Long.class));
```

Comment:
Can't do this in method `create`, because we don't create id for cars manually. So we don't use `car_id` in query and method getObject can't find this column.